src/lockjaw/protocol.clj

```diff
@@ -2,9 +2,15 @@
 
 (defprotocol Lockjaw
   (acquire! [this]
-    "Tries to get a lock for given ID")
+    "Tries to get a lock for given component ID")
+  (acquire-by-name! [this lock-name]
+    "Converts passed name to ID and tries to aquire a lock for it")
   (release! [this]
-    "Rleases the lock"))
+    "Rleases the component lock")
+  (release-by-name! [this lock-name]
+    "Converts passed name to ID and releases it")
+  (release-all! [this]
+    "Releases all acquired locks"))
 
 (defmacro with-lock
```

Comment:
These macros will need to be extended or we will need new ones (`with-named-lock` or something like that) so that the component is easier to use